extensions/wikibase/src/org/openrefine/wikibase/editing/MediaFileUtils.java

```diff
@@ -77,8 +78,31 @@ public void purgePage(long pageid) throws IOException, MediaWikiApiErrorExceptio
         Map<String, String> parameters = new HashMap<>();
         parameters.put("action", "purge");
         parameters.put("pageids", Long.toString(pageid));
-
-        apiConnection.sendJsonRequest("POST", parameters);
+        int retries = 5;
+        int backofTime = maxLagWaitTime;
+        while (retries > 0) {
+            try {
+                JsonNode response = apiConnection.sendJsonRequest("POST", parameters);
+                if (!(response != null && response.has("warnings"))) {
+                    return;
+                }
+            } catch (MediaWikiApiErrorException | IOException e) {
+            }
+            retries--;
+            if (retries > 0) {
+                backofTime = retries == 4 ? maxLagWaitTime : maxLagWaitTime * backoffFactor;
```

Comment:
This means that we are going to sleep for `maxLagWaitTime` the first time and then `maxLagWitTime * backoffFactor`. That's not quite what Wikidata-Toolkit does: an exponential backoff strategy, where the wait time will increase by a fixed factor at each new attempt.
The reason why this strategy is preferable is that it helps the web service reach a load equilibrium. You can read more about this here: https://en.wikipedia.org/wiki/Exponential_backoff#Rate_limiting
You can see how it is implemented in Wikidata-Toolkit:
https://github.com/Wikidata/Wikidata-Toolkit/blob/a9b1fb4d4cc36d7441d9a7e6cc5af02979bd3da1/wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WbEditingAction.java#L743-L764

I'd like to use the same strategy here for consistency.

Reply:
I will check this